### ISSUE TITLE
Confusing unexpected parsing error.

### ISSUE THREAD

**Issue by cschuller** [zeitport/eslint-plugin-json#7]
### Problem
Given invalid JSON like

```
{
```

or 

```
{"foo": [}}
```

Then the plugin returns this message:

```
1:1 Parsing error: Unexpected token )
```

### Explanation

Internally the given JSON `{` will be transformed to a statement `({)`.
The statement is not correct and can not be parsed by the espree (AST) parser and a generic parse error is thrown.
Custom rules are not called after a generic parser error.
